### PR TITLE
add safe classes

### PR DESCRIPTION
**Pull request by MoLow**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -2363,6 +2373,152 @@ const processHasUncaughtExceptionCaptureCallback = staticCall(
...
+          SymbolIterator in
+            (FunctionPrototypeCall(desc.value, dummy) ?? {})
+        ) {
+          const x: string[] = []
+          const y = x[Symbol.iterator]
+
+          const createIterator = uncurryThis(desc.value) as unknown as (
+            val: T
+          ) => IterableIterator<any>
+          next ??= uncurryThis(createIterator(dummy).next)
```
Comment (isaacs) [inline src/index.ts]: Not supported on node 14, so this is a breaking change. It's a function, so `||=` is equivalent.
```suggestion
          next ||= uncurryThis(createIterator(dummy).next)
```

**Review by isaacs — CHANGES_REQUESTED** [review]
Looks mostly fine, but tests aren't complete yet.

```
----------|---------|----------|---------|---------|-----------------------------------------------------------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                                                     
----------|---------|----------|---------|---------|-----------------------------------------------------------------------
All files |   99.09 |      100 |    82.5 |   99.09 |                                                                       
 index.ts |   99.09 |      100 |    82.5 |   99.09 | 2381-2394,2439-2440,2470-2471,2486-2487,2495-2496,2503-2504,2512-2518 
----------|---------|----------|---------|---------|-----------------------------------------------------------------------
```

**Comment by MoLow** [comment]
@isaacs can we remove node 14 from CI? it is EOL

File: src/index.ts [hunk 2]
```diff
@@ -2363,6 +2373,152 @@ const processHasUncaughtExceptionCaptureCallback = staticCall(
...
+    if (!ReflectGetOwnPropertyDescriptor(dest, key)) {
+      ReflectDefineProperty(dest, key, {
+        __proto__: null,
+        ...ReflectGetOwnPropertyDescriptor(src, key),
+      } as PropertyDescriptor)
+    }
+  })
+}
+
+interface Constractable<T> extends NewableFunction {
```
Comment (isaacs) [inline src/index.ts]: `s/Constractable/Constructable/`?

**Pull request closed without merging**